lib/src/accordion/Accordion.tsx

```diff
@@ -24,54 +32,73 @@ const DxcAccordion = ({
   padding,
   tabIndex = 0,
 }: AccordionPropsType): JSX.Element => {
+  const id = useId();
   const [innerIsExpanded, setInnerIsExpanded] = useState(defaultIsExpanded ?? false);
   const colorsTheme = useTheme();
 
-  const handlerAccordion = () => {
-    if (isExpanded == null) {
-      setInnerIsExpanded(!innerIsExpanded);
-    }
-    if (typeof onChange === "function") {
-      onChange(isExpanded == null ? !innerIsExpanded : !isExpanded);
-    }
+  const handleAccordionState = () => {
+    isExpanded ?? setInnerIsExpanded((innerIsExpanded) => !innerIsExpanded);
+    onChange?.(isExpanded == null ? !innerIsExpanded : !isExpanded);
   };
 
   return (
     <ThemeProvider theme={colorsTheme.accordion}>
-      <DXCAccordion padding={padding} margin={margin} disabled={disabled} icon={icon}>
-        <ExpansionPanel
-          disabled={disabled}
-          onChange={handlerAccordion}
-          expanded={isExpanded != null ? isExpanded : innerIsExpanded}
-        >
-          <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />} tabIndex={disabled ? -1 : tabIndex}>
+      <AccordionContainer isExpanded={isExpanded ?? innerIsExpanded} padding={padding} margin={margin}>
+        <AccordionHeader>
+          <AccordionTrigger
+            id={`accordion-${id}`}
+            onClick={disabled ? undefined : handleAccordionState}
+            disabled={disabled}
+            tabIndex={disabled ? -1 : tabIndex}
+            aria-expanded={isExpanded ?? innerIsExpanded}
+            aria-controls={`accordion-panel-${id}`}
+            isExpanded={isExpanded ?? innerIsExpanded}
+          >
             <AccordionInfo disabled={disabled}>
-              {icon && (
-                <IconContainer disabled={disabled}>
-                  {typeof icon === "string" ? <AccordionIcon src={icon} /> : icon}
-                </IconContainer>
-              )}
-              <AccordionLabel>{label}</AccordionLabel>
+              <AccordionLabel>
+                {icon && (
+                  <IconContainer disabled={disabled}>
+                    {typeof icon === "string" ? <IconImg src={icon} /> : icon}
+                  </IconContainer>
+                )}
+                {label}
+              </AccordionLabel>
+              {assistiveText && <AccordionAssistiveText disabled={disabled}>{assistiveText}</AccordionAssistiveText>}
```

Comment:
We could also use DxcTypography for the assistiveText.

Reply:
Done!